ledger/simulation/simulation_eval_test.go

```diff
@@ -1475,6 +1573,570 @@ int 1`
+			binary.BigEndian.PutUint64(bytes, v)
+			return bytes
+		}
+
+		creationOpcodeTrace := []simulation.OpcodeTraceUnit{
+			{PC: 1},
+			{PC: 6},
+			{PC: 8},
+			{PC: 9},
+			{PC: 10},
+			{PC: 185},
+			{PC: 186},
+		}
+
+		clearStateOpcodeTrace := []simulation.OpcodeTraceUnit{{PC: 1}}
+
+		recursiveLongOpcodeTrace := []simulation.OpcodeTraceUnit{
+			{PC: 1},
+			{PC: 6},
+			{PC: 8},
+			{PC: 9},
+			{PC: 10},
+			{PC: 13},
+			{PC: 15},
+			{PC: 16},
+			{PC: 17},
+			{PC: 20},
+			{PC: 22},
+			{PC: 23},
+			{PC: 24},
+			{PC: 28},
+			{PC: 30},
+			{PC: 32},
+			{PC: 34},
+			{PC: 36},
+			{PC: 38},
+			{PC: 40},
+			{PC: 42},
+			{PC: 44},
+			{PC: 46},
+			{PC: 48},
+			{PC: 50},
+			{PC: 52},
+			{PC: 54},
+			{PC: 55},
+			{PC: 57},
+			{PC: 58},
+			{PC: 60},
+			{PC: 61},
+			{PC: 63},
+			{PC: 66},
+			{PC: 67},
+			{PC: 68},
+			{PC: 69},
+			{PC: 70},
+			{PC: 73},
+			{PC: 74},
+			{PC: 75},
+			{PC: 76},
+			{PC: 81},
+			{PC: 82},
+			{PC: 83},
+			{PC: 85},
+			{PC: 86},
+			{PC: 88},
+			{PC: 90},
+			{PC: 92},
+			{PC: 94},
+			{PC: 96},
+			{PC: 97},
+			{PC: 98},
+			{PC: 99},
+			{PC: 101},
+			{PC: 102},
+			{PC: 104},
+			{PC: 106},
+			{PC: 110},
+			{PC: 111},
+			{PC: 113},
+			{PC: 120},
+			{PC: 123},
+			{PC: 124},
+			{PC: 125},
+			{PC: 126},
+			{PC: 128},
+			{PC: 129},
+			{PC: 130},
+			{PC: 132},
+			{PC: 134},
+			{PC: 136},
+			{PC: 137},
+			{PC: 139},
+			{PC: 140},
+			{PC: 142},
+			{PC: 143},
+			{PC: 144},
+			{PC: 146},
+			{PC: 148},
+			{PC: 150},
+			{PC: 151},
+			{PC: 153},
+			{PC: 155},
+			{PC: 157},
+			{PC: 158},
+			{PC: 159},
+			{PC: 161},
+			{PC: 164},
+			{PC: 165},
+			{PC: 166},
+			{PC: 167},
+			{PC: 168},
+			{PC: 170},
+			{PC: 172},
+			{PC: 174},
+			{PC: 175},
+			{PC: 177},
+			{PC: 179},
+			{PC: 181},
+			{PC: 182},
+			{PC: 79},
+			{PC: 80},
+		}
+
+		optInTrace := []simulation.OpcodeTraceUnit{
+			{PC: 1},
+			{PC: 6},
+			{PC: 8},
+			{PC: 9},
+			{PC: 10},
+			{PC: 13},
+			{PC: 15},
+			{PC: 16},
+			{PC: 17},
+			{PC: 185},
+			{PC: 186},
+		}
+
+		finalDepthTrace := []simulation.OpcodeTraceUnit{
+			{PC: 1},
+			{PC: 6},
+			{PC: 8},
+			{PC: 9},
+			{PC: 10},
+			{PC: 13},
+			{PC: 15},
+			{PC: 16},
+			{PC: 17},
+			{PC: 20},
+			{PC: 22},
+			{PC: 23},
+			{PC: 24},
+			{PC: 28},
+			{PC: 30},
+			{PC: 32},
+			{PC: 34},
+			{PC: 36},
+			{PC: 38},
+			{PC: 40},
+			{PC: 42},
+			{PC: 44},
+			{PC: 46},
+			{PC: 48},
+			{PC: 50},
+			{PC: 52},
+			{PC: 54},
+			{PC: 55},
+			{PC: 57},
+			{PC: 58},
+			{PC: 60},
+			{PC: 61},
+			{PC: 63},
+			{PC: 66},
+			{PC: 67},
+			{PC: 68},
+			{PC: 69},
+			{PC: 70},
+			{PC: 73},
+			{PC: 74},
+			{PC: 75},
+			{PC: 76},
+			{PC: 79},
+			{PC: 80},
+		}
+
+		stepToInnerMap := []simulation.TraceStepInnerIndexPair{
+			{TraceStep: 51, InnerIndex: 0},
+			{TraceStep: 100, InnerIndex: 1},
+			{TraceStep: 100, InnerIndex: 2},
+			{TraceStep: 100, InnerIndex: 3},
+			{TraceStep: 100, InnerIndex: 4},
```

Comment:
Why do these all have the same `TraceStep`?

Reply:
they are deriving from the second `itxn_submit` from PC 100, so there are 4 inners, but spawning point is PC 100.